switchboard/dds_util.py

```diff
@@ -524,11 +524,12 @@ def transfer_delivery(delivery, user):
         sender_message = message_factory.make_processed_message('accepted',
                                                                 MessageDirection.ToSender,
                                                                 warning_message=warning_message)
-        sender_message.send()
         recipient_message = message_factory.make_processed_message('accepted_recipient',
                                                                    MessageDirection.ToRecipient)
-        recipient_message.send()
+        # Save email messages first so the the emails can be resent if they fail
         delivery.mark_accepted(user.get_username(), sender_message.email_text, recipient_message.email_text)
+        sender_message.send()
+        recipient_message.send()
```

Comment:
We should catch the timeout exception raised in this two methods and add it to the warning message. Relevant docs:  
https://docs.python.org/3/library/smtplib.html#smtplib.SMTP.
https://docs.python.org/3/library/socket.html#socket.timeout